terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -81,6 +88,58 @@ def update_registry_declaration(new_req, old_req, updated_content)
         end
       end
 
+      def update_lockfile_declaration(dependency)
+        @updated_lockfile_declarations ||= {}
+        @updated_lockfile_declarations[dependency] ||=
+          begin
+            new_req = dependency.requirements.first
+            lockfile = lock_file.first
+            content = lockfile.content.dup
+
+            provider_source = new_req[:source][:registry_hostname] + "/" + new_req[:source][:module_identifier]
+            declaration_regex = lockfile_declaration_regex(provider_source)
+            lockfile_dependency_removed = content.sub(declaration_regex, "")
+
+            SharedHelpers.in_a_temporary_directory do
+              write_dependency_files
+
+              File.write(".terraform.lock.hcl", lockfile_dependency_removed)
+              SharedHelpers.run_shell_command("terraform providers lock #{provider_source}")
+
+              updated_lockfile = File.read(".terraform.lock.hcl")
+              updated_dependency = updated_lockfile.scan(declaration_regex).first
+
+              # Terraform will occasionally update h1 hashes without updating the version of the dependency
+              # Here we make sure the dependency's version actually changes in the lockfile
```

Comment:
This comment implies this only happens occasionally but it looks like we only update the original lockfile content if the version has changed in the updated lockfile. Have I interpreted this correctly?

```suggestion
              # Update the original lockfile content if the version of the dependency has changed in the updated 
              # lockfile along with the h1 hashes. This prevents updating unrelated dependencies.
```

Reply:
From the [dependency lock docs](https://www.terraform.io/docs/language/dependency-lock.html):

> Due to the limited scope of the zh: scheme, Terraform will opportunistically add in the corresponding h1: checksums as it learns of them

You are correct that we only update the original lockfile content if the version has changed in the updated lockfile. However, the reason the check is there is to prevent dependabot from updating the lockfile hash when the version has not changed.